llama_index/schema.py

```diff
@@ -119,3 +147,303 @@ def extra_info_str(self) -> Optional[str]:
+            raise ValueError("Object does not have next link")
+        if not isinstance(self.relationships[DataRelationship.NEXT], str):
+            raise ValueError("Next object must be a string id")
+        return self.relationships[DataRelationship.NEXT]
+
+    @property
+    def parent_object_id(self) -> str:
+        """Parent object id."""
+        if DataRelationship.PARENT not in self.relationships:
+            raise ValueError("Object does not have parent link")
+        if not isinstance(self.relationships[DataRelationship.PARENT], str):
+            raise ValueError("Parent object must be a string id")
+        return self.relationships[DataRelationship.PARENT]
+
+    @property
+    def child_object_ids(self) -> List[str]:
+        """Child object ids."""
+        if DataRelationship.CHILD not in self.relationships:
+            raise ValueError("Object does not have child objects")
+        if not isinstance(self.relationships[DataRelationship.CHILD], list):
+            raise ValueError("Child objects must be a list ids")
+        return self.relationships[DataRelationship.CHILD]
+
+    @property
+    def extra_info(self) -> Dict[str, Any]:
+        return self.metadata
+
+
+@dataclass
+class BaseNode(BaseIndexObject):
+    content: str = ""
+
+    def _generate_object_hash(self) -> str:
+        """Generate a hash to represent the object."""
+        doc_identity = str(self.content) + str(self.metadata_str)
+        return sha256(doc_identity.encode("utf-8", "surrogatepass")).hexdigest()
+
+    @classmethod
+    def get_type(cls) -> str:
+        """Get Object type."""
+        return ObjectType.TEXT
+
+    def get_content(self) -> str:
+        """Get object content."""
+        return (f"{self.metadata_str}\n\n" f"{self.content}").strip()
+
+    def get_node_info(self) -> Dict[str, Any]:
+        """DEPRECATED: Get node info."""
+        if self.attributes is None:
+            raise ValueError("Node attributes not set.")
+        return self.attributes
+
+    def get_text(self) -> str:
+        """DEPRECATED: Get text."""
+        return self.get_content()
+
+    def get_origin_type(self) -> str:
+        """Get origin type."""
+        if self.attributes is None or "_node_type" not in self.attributes:
+            return self.get_type()
+        return self.attributes["_node_type"]
+
+    # TODO: deprecated node properties
+    @property
+    def ref_doc_id(self) -> Optional[str]:
+        return self.source_object_id
+
+    @property
+    def prev_node_id(self) -> str:
+        return self.prev_object_id
+
+    @property
+    def next_node_id(self) -> str:
+        return self.next_object_id
+
+    @property
+    def parent_node_id(self) -> str:
+        return self.parent_object_id
+
+    @property
+    def child_node_ids(self) -> List[str]:
+        return self.child_object_ids
+
+    @property
+    def node_info(self) -> Dict[str, Any]:
+        return self.attributes
+
+
+@dataclass
+class Document(BaseNode):
```

Comment:
could you remind me how the Document itself can be used + how would title and description be used in underlying nodes?

Reply:
llama index do not document concept for now, add document object would make it easier to refresh documents and more intuitive for code in index building and reference info building such like add reference document object as a property to a node

Reply:
I added a `RefNodeInfo` class to explicitly track this better (which will also fix the `ref_doc_info` support for vector stores!)